src/IO/ReadHelpers.cpp

```diff
@@ -700,16 +701,18 @@ void readCSVStringInto(Vector & s, ReadBuffer & buf, const FormatSettings::CSV &
             if (!buf.hasPendingData())
                 continue;
 
-            /** CSV format can contain insignificant spaces and tabs.
-              * Usually the task of skipping them is for the calling code.
-              * But in this case, it will be difficult to do this, so remove the trailing whitespace by ourself.
-              */
-            size_t size = s.size();
-            while (size > 0
-                && (s[size - 1] == ' ' || s[size - 1] == '\t'))
-                --size;
-
-            s.resize(size);
+            if constexpr (!std::is_same_v<Vector, NullOutput>)
```

Comment:
You can make this check a bit more general with concepts:
```
  template<typename T>
  concept WithResize = requires (T value)
  {
      { value.resize(1) };
      { value.size() } -> std::integral<>;
  };
  ```
and then do
```suggestion
            if constexpr (WithResize<Vector>)
```